### PR TITLE
DBAAS-1170: workaround to the rds-controller upgrade issue for release 0.0.4

### PR DESCRIPTION
**Pull request by xieshenzh**
## Description

The aws `rds-controller` requires a configuration configmap and a credentials secret to finish install and upgrade.
The `rds-dbaas-operator` will scale it down to avoid this issue when installation, but during upgrade the `rds-dbaas-operator` is deleted. `rds-controller` will auto-upgrade,`rds-dbaas-operator` won't be installed until the `rds-controller` upgrade finishes or deleted. As a result, the upgrade will be in a dead lock if there is no rds provider account before upgrading.

The workaround is to create dummy aws secret and configmap if not exist during the `dbaas-operator` controllers installation. `ack-controller` and `rds-dbaas-operator` start upgrading after the `dbaas-operator` installation finishes.

## Verification Steps

Tested 3 scenarios:
1. fresh installation, create provider account for RDS, then create provider account for RDS
2. install v0.0.3, create provider account for RDS on v0.0.3, then upgrade to v0.0.4
3. install v0.0.3, not create provider account for RDS, then upgrade to v0.0.4 and create provider account for RDS

## Type of change

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist (Definition of Done)

- [ ] All acceptance criteria specified in JIRA or in issue number have been completed
- [ ] Unit tests added that prove the fix is effective or the feature works 
- [ ] Documentation added for the feature
- [ ] all relevant tests are passing
- [ ] Code Review completed
- [ ] Verified independently by reviewer

### CODE REVIEW

**Comment by xieshenzh** [comment]
Overall coverage for the file is ~50%, but for the new code I added it is 73.3% . 

image

**Review by tchughesiv — COMMENTED** [review]
thanks for the contrib! I have concerns about memory caching/leaks related to configmaps & secrets. Those are the 2 biggest offenders and we've had to take precaution to avoid CM watches in the past. With regard to secrets, while we do watch them... we take precautions like the following -
https://github.com/RHEcosystemAppEng/dbaas-operator/blob/346d6c647ba154d9c44effdfd4c0fd7abb726a5e/main.go#L124-L127

**Comment by xieshenzh** [comment]
> thanks for the contrib! I have concerns about memory caching/leaks related to configmaps & secrets. Those are the 2 biggest offenders and we've had to take precaution to avoid CM watches in the past. With regard to secrets, while we do watch them... we take precautions like the following -
> 
> https://github.com/RHEcosystemAppEng/dbaas-operator/blob/346d6c647ba154d9c44effdfd4c0fd7abb726a5e/main.go#L124-L127
> 
> I worry, with a change like this... we'd have to worry about memory leaks again or, at the very least... this would prompt some intensive performance testing/sessions.

Are you suggesting we remove the watch permission and disable cache for configmap as well?

**Comment by tchughesiv** [comment]
@xieshenzh at the very least, yes. we'll also need broader team discussion

**Comment by tchughesiv** [comment]
@xieshenzh probably don't need the `list` perm either

File: main.go [hunk 1]
```diff
@@ -124,6 +124,7 @@ func main() {
 		ClientDisableCacheFor: []client.Object{
 			&operatorframework.ClusterServiceVersion{},
 			&corev1.Secret{},
+			&corev1.ConfigMap{},
```
Comment (tchughesiv) [inline main.go:127]: please add this to the test suites as well...
here - 
https://github.com/RHEcosystemAppEng/dbaas-operator/blob/346d6c647ba154d9c44effdfd4c0fd7abb726a5e/controllers/suite_test.go#L124-L127
and here -
https://github.com/RHEcosystemAppEng/dbaas-operator/blob/346d6c647ba154d9c44effdfd4c0fd7abb726a5e/api/v1beta1/webhook_suite_test.go#L116-L119

**Review by tchughesiv — CHANGES_REQUESTED** [review]

File: main.go (hunk 1 above)
Comment (xieshenzh) [reply main.go:127]: Done

**Review by tchughesiv — APPROVED** [review]
lgtm

**Pull request merged**